plugins/crate-copy-azure/src/main/java/io/crate/copy/azure/AzureBlobStorageSettings.java

```diff
@@ -0,0 +1,94 @@
+/*
+ * Licensed to Crate.io GmbH ("Crate") under one or more contributor
+ * license agreements.  See the NOTICE file distributed with this work for
+ * additional information regarding copyright ownership.  Crate licenses
+ * this file to you under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.  You may
+ * obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.  See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ *
+ * However, if you have executed another commercial license agreement
+ * with Crate these terms will supersede the license and you may use the
+ * software solely pursuant to the terms of the relevant commercial agreement.
+ */
+
+package io.crate.copy.azure;
+
+import java.util.HashMap;
+import java.util.List;
+import java.util.Locale;
+import java.util.Map;
+import java.util.Set;
+import java.util.stream.Collectors;
+
+import org.elasticsearch.common.settings.Setting;
+import org.elasticsearch.common.settings.Setting.Property;
+import org.elasticsearch.common.settings.Settings;
+
+/**
+ * Reflects settings from https://docs.rs/opendal/latest/opendal/services/struct.Azblob.html
+ */
+public class AzureBlobStorageSettings {
+
+    // All operations will happen under this directory (user provided path will be prepended by root).
+    static final Setting<String> ROOT_SETTING = Setting.simpleString("root", Property.NodeScope);
+
+    static final Setting<String> ACCOUNT_NAME_SETTING = Setting.simpleString("account_name", Property.NodeScope);
+    static final Setting<String> ACCOUNT_KEY_SETTING = Setting.simpleString("account_key", Property.NodeScope);
+
+    static final Setting<String> CONTAINER_SETTING = Setting.simpleString("container", Property.NodeScope);
+    static final Setting<String> ENDPOINT_SETTING = Setting.simpleString("endpoint", Property.NodeScope);
+
+    /**
+     * Registry of all settings also containing information whether setting is required or not.
+     */
+    private static Map<Setting<String>, Boolean> supportedSettings() {
+        return Map.of(
+            CONTAINER_SETTING, true,
+            ENDPOINT_SETTING, true,
+            ACCOUNT_NAME_SETTING, true,
+            ACCOUNT_KEY_SETTING, true,
+            ROOT_SETTING, false
+        );
+    }
+
+    /**
+     * Creates OpenDAL config from user provided settings.
+     * Rejects unknown settings.
+     * @param commonCopySettings is either common COPY TO or common COPY FROM settings.
+     */
+    public static Map<String, String> openDALConfig(Settings settings, List<String> commonCopySettings) {
```

Comment:
Also, why do we need to pass in here `commonCopySettings` if they are addressable in a static way anyhow?

Reply:
>can be done where it is needed

I introduced this because we need it in 2 places (FileInput/FileOutput). Will split it up to avoid confusion.

> why do we need to pass in here commonCopySettings
it's wither COPY_TO, or COPY_FROM, depending on entry point (FileInput/FileOutput). But yeah, doesn't make sense to pass static, will do smth else.

Reply:
I added azure->opendal names rewriting in https://github.com/crate/crate/pull/16468/commits/b8d7ffe260efa4486f05ef894c1a4bfaf2fb529c

Feel like we actually need 1 method to build OpenDAL config.
Also, will be able to get rid of one test (https://github.com/crate/crate/pull/16468#discussion_r1757073329)